### PR TITLE
Laurislopata/edit profile

### PR DESCRIPTION
**Pull request by laurislopata**

### CODE REVIEW

**Comment by laurislopata** [comment]
In this PR I implemented google sign in with the tests I could, added email and password validation and implemented username and handle change

**Review by Tsathogguaa — APPROVED** [review]
Multiple warnings about using android resources and hardcoding strings. We already have quite a few things to fix and will fix these soon as well, however in the future please pay attention to remove as many of these as possible. Android studio's "Problems" tab at the bottom shows both errors and warnings, which might be useful.

File: app/src/main/java/ch/sdp/vibester/Register.kt [hunk 1]
```diff
@@ -27,61 +27,79 @@ class Register : AppCompatActivity() {
...
+        }
+    }
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        updateUI(authenticator.googleActivityResult(requestCode, resultCode, data));
+    }
+
+    private fun stringValidation(username: String, password: String): Boolean{
+        if(username.isEmpty() || password.isEmpty()) {
+            email.text = "Empty email or password"
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/Register.kt:72]: May be using string.xml

File: app/src/main/java/ch/sdp/vibester/Register.kt [hunk 2]
```diff
@@ -16,7 +16,7 @@ import com.google.firebase.auth.AuthResult
 
 class Register : AppCompatActivity() {
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/Register.kt:17]: May add javadoc to class and methods ! Maybe some comments too

File: app/src/main/java/ch/sdp/vibester/auth/FireBaseAuthenticator.kt [hunk 3]
```diff
@@ -31,18 +31,17 @@ class FireBaseAuthenticator() {
...
-//                account.email
-//            } catch (e: ApiException) {
-//                "Authentication error"
-//            }
-//        } else {
-//            "Authentication error"
-//        }
-//    }
-
+    fun googleActivityResult(requestCode: Int, resultCode: Int, data: Intent?): String? {
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/auth/FireBaseAuthenticator.kt:34]: doc ? javadoc ?

File: app/src/main/java/ch/sdp/vibester/profile/ProfileDataProvider.kt [hunk 4]
```diff
@@ -2,7 +2,7 @@ package ch.sdp.vibester.profile
 
 class ProfileDataProvider(userID: String, users: List<UserProfile> = emptyList(), scoreboard: List<Int> = emptyList()) {
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/profile/ProfileDataProvider.kt:3]: doc ?

**Review by MaximeZmt — COMMENTED** [review]
Good code in overall 👍 , just missing some javadoc maybe 😅

File: app/src/main/java/ch/sdp/vibester/auth/FireBaseAuthenticator.kt [hunk 5]
```diff
@@ -31,18 +31,17 @@ class FireBaseAuthenticator() {
...
+        return if(requestCode == 1000) {
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/auth/FireBaseAuthenticator.kt:35]: maybe replace this magic number by a more meaningful name

File: app/src/androidTest/java/ch/sdp/vibester/SignInActivityTest.kt [hunk 6]
```diff
@@ -38,38 +38,83 @@ class SignInActivityTest {
...
+        onView(withId(R.id.password)).perform(ViewActions.typeText(password), closeSoftKeyboard())
+        onView(withId(R.id.createAcc)).perform(click())
+        onView(withId(R.id.email)).check(matches(withText("Password has to be at least 6 symbols")))
+    }
+
+    @Test
+    fun derinTest() {
+        val a = FireBaseAuthenticator()
+        a.googleActivityResult(-1,-1,null)
+        onView(withId(R.id.email)).check(matches(withText("TextView")))
```
Comment (jiabaow) [inline app/src/androidTest/java/ch/sdp/vibester/SignInActivityTest.kt:99]: Why do we need a "TextView" here, is it shown explicitly on the screen?

**Review by jiabaow — APPROVED** [review]
I added some comments but the rest LGTM

**Comment by laurislopata** [comment]
Good comments, I'll merge this and fix them with my next PR. I want to have google Sign In in our demo just in case I don't finish my other task

**Pull request merged**